src/components/Course/Placeholder.vue

```diff
@@ -15,17 +26,48 @@
 <script lang="ts">
 import { PropType, defineComponent } from 'vue';
 import CourseCaution from '@/components/Course/CourseCaution.vue';
+import NewSelfCheckOrPlaceholderCourseModal from '@/components/Modals/NewCourse/NewSelfCheckOrPlaceholderCourseModal.vue';
 
 export default defineComponent({
-  components: { CourseCaution },
+  components: { CourseCaution, NewSelfCheckOrPlaceholderCourseModal },
   props: {
     placeholderObj: { type: Object as PropType<FirestoreSemesterPlaceholder>, required: true },
     compact: { type: Boolean, required: true },
     semesterIndex: { type: Number, required: false, default: 0 },
   },
+  emits: {
+    'delete-placeholder': (placeholderObj: FirestoreSemesterPlaceholder) =>
+      typeof placeholderObj === 'object',
+  },
+  data() {
+    return {
+      isPlaceholderModalOpen: false,
+    };
+  },
   methods: {
     openModal() {
-      // TODO: open modal to assign a course to a placeholder
+      this.isPlaceholderModalOpen = true;
+    },
+    closeModal() {
+      this.isPlaceholderModalOpen = false;
+    },
+    assignCourse() {
+      // TODO @willespencer write the code that assigns a course to a placeholder
+    },
+    getRequirementID() {
+      // TODO @willespencer get requirement ID from name and slot
```

Comment:
Just double checking, do placeholders and requirements have a 1:1 mapping? Where is the name and slot stored?

Reply:
Requirement name and slot is in the firestore list of courses! Every requirement slot will have at most as many placeholders as courses that are needed to fulfill it, but could have fewer in cases of double counting.